files/initramfs-tools/union-mount.j2

```diff
@@ -39,10 +39,38 @@ set_tmpfs_log_partition_size()
   [ $maxsize -le $varlogsize ] && varlogsize=$maxsize
 }
 
+whitelist_rw_folder()
+{
+  image_dir=$1
+  whitelist_file=${rootmnt}/host/$image_dir/whitelist_paths.conf
+  
+  # Return if the whitelist file does not exist
+  if ! test -f "${whitelist_file}"; then
+    return
+  fi
+
+  # Return if the secure_boot_enable option is not set
+  if ! (cat /proc/cmdline | grep -i -q "secure_boot_enable=[y1]"); then
+    return
+  fi
+
+  rw_dir=${rootmnt}/host/$image_dir/rw
+
+  # Set the grep pattern file
+  whitelist_pattern_file=${rootmnt}/host/$image_dir/whitelist_paths.pattern
+  grep -v "^\s*$" ${whitelist_file} | awk -v rw_dir="$rw_dir" '{print rw_dir"/"$0"$"}' > $whitelist_pattern_file
```

Comment:
>awk [](start = 38, length = 3)

Can we use awk to handle empty line [https://stackoverflow.com/a/11687266/2514803](https://stackoverflow.com/a/11687266/2514803)
And remove grep.

Reply:
Fixed.